parts/linux/cloud-init/artifacts/cse_main.sh

```diff
@@ -400,17 +400,32 @@ function nodePrep {
             logs_to_events "AKS.CSE.ensureMigPartition" ensureMigPartition
         fi
 
+        # Install GPU device plugin if needed and not already installed
+        if [ "${ENABLE_GPU_DEVICE_PLUGIN_IF_NEEDED}" = "true" ]; then
+            if ! systemctl list-unit-files | grep -q "nvidia-device-plugin.service"; then
+                echo "Installing nvidia-device-plugin package..."
+                logs_to_events "AKS.CSE.installNvidiaDevicePlugin" "installNvidiaDevicePluginPkgFromCache"
+            else
+                echo "nvidia-device-plugin package already installed"
+            fi
+        fi
+
         # Configure GPU device plugin
         if [ "${ENABLE_GPU_DEVICE_PLUGIN_IF_NEEDED}" = "true" ]; then
-            if [ "${MIG_NODE}" = "true" ] && [ -f "/etc/systemd/system/nvidia-device-plugin.service" ]; then
-                mkdir -p "/etc/systemd/system/nvidia-device-plugin.service.d"
+            # Ensure kubelet device-plugins directory exists
+            mkdir -p /var/lib/kubelet/device-plugins
+            
+            if [ "${MIG_NODE}" = "true" ]; then
                 tee "/etc/systemd/system/nvidia-device-plugin.service.d/10-mig_strategy.conf" > /dev/null <<'EOF'
 [Service]
 Environment="MIG_STRATEGY=--mig-strategy single"
 ExecStart=
-ExecStart=/usr/local/nvidia/bin/nvidia-device-plugin $MIG_STRATEGY
+ExecStart=/usr/bin/nvidia-device-plugin $MIG_STRATEGY
```

Comment:
Correct me if I am wrong here, but I think that the `MIG_STRATEGY` is being populated from the line # 421 with value `--mig-strategy single`, so no matter what the MIG strategy is, it is always gonna be single.

Reply:
That's desired. We are not supporting mixed since there's no way to specify that for users today.